pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -192,16 +192,22 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 		return errors.Wrap(err, "error while trying to fetch service(s) from devfile")
 	}
 	labels := componentlabels.GetLabels(a.ComponentName, a.AppName, true)
-	// create the Kubernetes objects from the manifest
-	services, err := service.CreateServiceFromKubernetesInlineComponents(a.Client.GetKubeClient(), k8sComponents, labels)
+	// create the Kubernetes objects from the manifest and delete the ones not in the devfile
+	createdServices, deletedServices, err := service.PushServiceFromKubernetesInlineComponents(a.Client.GetKubeClient(), k8sComponents, labels)
 	if err != nil {
 		return errors.Wrap(err, "failed to create service(s) associated with the component")
 	}
 
-	if len(services) == 1 {
-		log.Infof("Created service %q on the cluster; refer %q to know how to link it to the component", services[0], "odo link -h")
-	} else if len(services) > 1 {
-		log.Infof("Created services %q on the cluster; refer %q to know how to link them to the component", strings.Join(services, ", "), "odo link -h")
+	if len(createdServices) == 1 {
+		log.Infof("Created service %q on the cluster; refer %q to know how to link it to the component", createdServices[0], "odo link -h")
+	} else if len(createdServices) > 1 {
+		log.Infof("Created services %q on the cluster; refer %q to know how to link them to the component", strings.Join(createdServices, ", "), "odo link -h")
+	}
+
+	if len(deletedServices) == 1 {
+		log.Infof("Deleted service %q from the cluster", deletedServices[0])
+	} else if len(deletedServices) > 1 {
+		log.Infof("Deleted services %q from the cluster", strings.Join(deletedServices, ", "))
```

Comment:
It might be a good idea to move these print statements inside `PushServiceFromKubernetesInlineComponents()` itself.